src/gui/guiFormSpecMenu.cpp

```diff
@@ -347,6 +348,111 @@ void GUIFormSpecMenu::parseContainerEnd(parserData* data)
 	}
 }
 
+void GUIFormSpecMenu::parseScrollContainer(parserData *data, const std::string &element)
+{
+	std::vector<std::string> parts = split(element, ';');
+
+	if (parts.size() < 4 ||
+			(parts.size() > 5 && m_formspec_version <= FORMSPEC_API_VERSION)) {
+		errorstream << "Invalid scroll_container start element (" << parts.size()
+				<< "): '" << element << "'" << std::endl;
+		return;
+	}
+
+	std::vector<std::string> v_pos  = split(parts[0], ',');
+	std::vector<std::string> v_geom = split(parts[1], ',');
+	std::string scrollbar_name = parts[2];
+	std::string direction = parts[3];
+	f32 scroll_factor = 0.1f;
+	if (parts.size() >= 5 && !parts[4].empty())
+		scroll_factor = stof(parts[4]);
+
+	MY_CHECKPOS("scroll_container", 0);
+	MY_CHECKGEOM("scroll_container", 1);
+
+	v2s32 pos = getRealCoordinateBasePos(v_pos);
+	v2s32 geom = getRealCoordinateGeometry(v_geom);
+
+	// old parent (at first: this)
+	// ^ is parent of clipper
+	// ^ is parent of mover
+	// ^ is parent of other elements
+
+	// make clipper
+	FieldSpec spec_clipper(
```

Comment:
Why do you need to allocate an ID for the clipper and mover?

Reply:
Afaik, this is done for every element.
What id should I use instead for the clipper?

Reply:
I admit that the scope of fieldspecs confuse me. To clarify - is there a reason why the id isn't -1 with no fieldspec?

Reply:
Hm, I don't think, anything would speak against -1.
Should I use -1?

Reply:
`FieldSpec` entries are only needed if the element....

* has a name (for styling, sending fields or tooltips)
* or needs special legacy sorting
* ~or has clipped text/label that is shown entirely using a tooltip~
   * ^ possible use-case in the future, not implemented.

All other `m_fields` entries are superfluous because they neither contain information nor provide any special functionality.